rust/chains/tw_solana/src/modules/message_builder.rs

```diff
@@ -71,7 +71,15 @@ impl<'a> MessageBuilder<'a> {
 
     pub fn build(self) -> SigningResult<VersionedMessage> {
         if let Some(ref raw_message) = self.input.raw_message {
-            return RawMessageBuilder::build(raw_message);
+            let mut result = RawMessageBuilder::build(raw_message);
+
+            if let Ok(version_message) = &mut result {
+                // Update the priority fee in raw message if priority fee is set in the signing input.
+                version_message
+                    .set_priority_fee(self.priority_fee_limit(), self.priority_fee_price());
+            }
```

Comment:
We shouldn't modify the given raw message hiddenly, but rather expose an API for that, such as `SolanaTransaction.setComputeUnitLimit()`, `SolanaTransaction.getComputeUnitPrice()`, `SolanaTransaction.setComputeUnitPrice()`.

Reply:
Hi @satoshiotomakan, do you mean we should implement updating the priority fee just like the SolanaTransaction.updateBlockhashAndSign?

Initially, I had implemented it in a similar way just liking SolanaTransaction.updateBlockhashAndSign, but I abandoned that approach due to the following reasons:

Reason 1: the "warning" comments in the code. 
https://github.com/trustwallet/wallet-core/blob/58fdabf62c17149862552da32f5b56a0aea21dae/include/TrustWalletCore/TWSolanaTransaction.h#L19-L29

Reason 2: the return value of SolanaTransaction.updateBlockhashAndSign is SigningOutput. It's not good for case of getting signature from signature server. In order to use function compileWithSignatures, user need construct SingingInput from this SigningOutput. It's a bit cumbersome.

Reason 3: the approach in this PR is very convenient for the users.
```
    let input = Proto::SigningInput {
         raw_message: Some(decoded_tx),
         priority_fee_limit: Some(Proto::PriorityFeeLimit { limit: 10_000 }),
         priority_fee_price: Some(Proto::PriorityFeePrice { price: 1_000 }),
         ..Proto::SigningInput::default()
     };
```
BTW, I think the method in this PR is not a **hidden** approach, as the priority_fee_limit and priority_fee_price settings in SigningInput are explicitly set by the user

Reply:
I totally agree on the `Reason 2`. The interface I plan to add will look like this:
* SolanaTransaction.getComputeUnitLimit(encodedTransaction: String) -> String?, where:
  * encodedTransaction - base64 encoded transaction
  * return - optional Unit Limit as a decimal string
* SolanaTransaction.setComputeUnitLimit(encodedTransaction: String, limit: String) -> String, where:
  * encodedTransaction - base64 encoded transaction
  * limit - decimal string
  * return - base64 encoded updated transaction
* SolanaTransaction.getComputeUnitPrice(encodedTransaction: String) -> String?, where
  * encodedTransaction - base64 encoded transaction
  * return - optional Unit Price as a decimal string
* SolanaTransaction.setComputeUnitPrice(encodedTransaction: String, price: String) -> String, where:
  * encodedTransaction - base64 encoded transaction
  * price - decimal string
  * return - base64 encoded updated transaction

So you'll get a serialized transaction back, and you can call `TransactionDecoder.decode()` to convert it to `Solana.Proto.RawMessage`

Reply:
Regarding `Reason 3` - probably it's a matter of taste. Anyway in TrustWallet we need to be able to get the previous `ComputeUnitLimit` and `ComputeUnitPrice` to compare them with our own estimations.
That's why we decided to add dedicated FFI functions